lib/engine/ability/token.rb

```diff
@@ -5,14 +5,15 @@
 module Engine
   module Ability
     class Token < Base
-      attr_reader :hexes, :price, :teleport_price, :extra, :from_owner
+      attr_reader :hexes, :price, :teleport_price, :extra, :from_owner, :discount
 
-      def setup(hexes:, price:, teleport_price: nil, extra: nil, from_owner: nil)
+      def setup(hexes:, price: nil, teleport_price: nil, extra: nil, from_owner: nil, discount: nil)
```

Comment:
why did you change price?

Reply:
In case if discount used there is no meaning of price, so there is no reasonable default value.
Should I add a guard/game_error in case both price and discount are nil?